documentation/optee_design.md

```diff
@@ -451,6 +451,36 @@ OP-TEE to be able to initialize at all. The loader supplies in `r0/x0` the
 address of the first byte following what was not copied and jumps to the load
 address to start OP-TEE.
 
+Extra image files are generated by loaders who support loading separate images.
+Legacy OP-TEE image tee.bin is split into several v2 images; typically header
+image, pager image and pageable image. Loaders load header image first to get
+image list and information of each image; and then load each of them into
+different load address assigned in structure; finally main image binary
+(`pager` in general) gets launched after all images loaded.
+Header image structure is updated to support separate loading:
+```c
+typedef struct optee_image {
```

Comment:
We don't like typedefs very much (see https://www.kernel.org/doc/html/v4.10/process/coding-style.html#typedefs). Please drop it and write `struct optee_image optee_image[];` below.